screens/onboarding/StoreSelectScreen.js

```diff
@@ -0,0 +1,204 @@
+import { FontAwesome5 } from '@expo/vector-icons';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { AsyncStorage, FlatList, View } from 'react-native';
+import { SearchBar } from 'react-native-elements';
+import {
+  Body,
+  ButtonContainer,
+  ButtonLabel,
+  FilledButtonContainer,
+  NavHeaderContainer,
+  Title,
+} from '../../components/BaseComponents';
+import StoreSelectCard from '../../components/store/StoreSelectCard';
+import Colors from '../../constants/Colors';
+import { getCustomersById, updateCustomers } from '../../lib/airtable/request';
+import { logErrorToSentry } from '../../lib/logUtils';
+import { ColumnContainer, RowContainer } from '../../styled/shared';
+import { styles } from '../../styled/store';
+
+export default class StoreSelectScreen extends React.Component {
+  constructor(props) {
+    super(props);
+    const {
+      navigation,
+      showDefaultStore,
+      updateStep,
+    } = this.props.route.params;
+    this.state = {
+      navigation,
+      searchStr: '',
+      showDefaultStore,
+      selectedStores: [],
+      updateStep,
+      isLoading: true,
+    };
+  }
+
+  async componentDidMount() {
+    const customerId = await AsyncStorage.getItem('customerId');
+    const cust = await getCustomersById(customerId);
+    const favoriteStores = cust.favoriteStoreIds || [];
+    this.setState({ selectedStores: favoriteStores });
+    this.setState({ isLoading: false });
+  }
+
+  selectStore = (storeId) => {
+    const index = this.state.selectedStores.indexOf(storeId);
+    this.setState((state) => {
+      let selectedStores;
+      if (index > -1) {
+        selectedStores = state.selectedStores.filter((_, i) => index !== i);
+      } else {
+        selectedStores = [...state.selectedStores, storeId];
+      }
+      return {
+        selectedStores,
+      };
+    });
+  };
+
+  filterStore = (searchStr) => {
```

Comment:
There's quite a bit of repetition between this file and the `StoreListScreen`, but does that warrant moving this out to a `utils` file?